grakn-engine/src/main/java/ai/grakn/engine/GraknEngineServerFactory.java

```diff
@@ -0,0 +1,96 @@
+ */
+
+package ai.grakn.engine;
+
+import ai.grakn.GraknConfigKey;
+import ai.grakn.engine.controller.HttpController;
+import ai.grakn.engine.data.RedisWrapper;
+import ai.grakn.engine.factory.EngineGraknTxFactory;
+import ai.grakn.engine.lock.JedisLockProvider;
+import ai.grakn.engine.lock.LockProvider;
+import ai.grakn.engine.rpc.GrpcServer;
+import ai.grakn.engine.task.BackgroundTaskRunner;
+import ai.grakn.engine.task.postprocessing.CountPostProcessor;
+import ai.grakn.engine.task.postprocessing.CountStorage;
+import ai.grakn.engine.task.postprocessing.IndexPostProcessor;
+import ai.grakn.engine.task.postprocessing.IndexStorage;
+import ai.grakn.engine.task.postprocessing.PostProcessingTask;
+import ai.grakn.engine.task.postprocessing.PostProcessor;
+import ai.grakn.engine.task.postprocessing.redisstorage.RedisCountStorage;
+import ai.grakn.engine.task.postprocessing.redisstorage.RedisIndexStorage;
+import ai.grakn.engine.util.EngineID;
+import com.codahale.metrics.MetricRegistry;
+import spark.Service;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * Static configurator for classes
+ *
+ * @author Michele Orsi
+ */
+public class GraknEngineServerFactory {
+    private static GraknEngineServer graknEngineServer;
+
+    public static synchronized GraknEngineServer getOrCreateGraknEngineServer() throws IOException {
+        if (graknEngineServer == null) {
+            graknEngineServer = createGraknEngineServer();
+        }
+        return graknEngineServer;
+    }
+
+    private static GraknEngineServer createGraknEngineServer() throws IOException {
+        GraknConfig config = GraknConfig.create();
+        RedisWrapper redisWrapper = RedisWrapper.create(config);
+        MetricRegistry metricRegistry = new MetricRegistry();
+        IndexStorage indexStorage =  RedisIndexStorage.create(redisWrapper.getJedisPool(), metricRegistry);
+        CountStorage countStorage = RedisCountStorage.create(redisWrapper.getJedisPool(), metricRegistry);
+        LockProvider lockProvider = new JedisLockProvider(redisWrapper.getJedisPool());
+        EngineGraknTxFactory engineGraknTxFactory = EngineGraknTxFactory.create(lockProvider, config);
+        return createGraknEngineServer(EngineID.me(), Service.ignite(), new GraknEngineStatus(), metricRegistry,
+                config, redisWrapper, indexStorage, countStorage, lockProvider, Runtime.getRuntime(), Collections.emptyList(), engineGraknTxFactory);
+    }
+
+    public static synchronized GraknEngineServer createGraknEngineServer(
+            EngineID engineID, Service sparkService, GraknEngineStatus graknEngineStatus, MetricRegistry metricRegistry, GraknConfig graknEngineConfig,
+            RedisWrapper redisWrapper, IndexStorage indexStorage, CountStorage countStorage, LockProvider lockProvider, Runtime runtime, Collection<HttpController> collaborators, EngineGraknTxFactory factory) throws IOException {
```

Comment:
This method signature is pretty nuts. Maybe at some point we should make this a builder? Not for this PR tho.

Reply:
good idea, i'll do that in the following PR when i got time